lua/neotest-haskell/init.lua

```diff
@@ -59,12 +59,16 @@ function HaskellNeotestAdapter.build_spec(args)
   end
 
   local package_root = base.match_package_root_pattern(pos.path)
-  local project_root = base.match_project_root_pattern(pos.path)
+  -- If this isn't a stack project or a multi-package cabal project,
+  -- then the root is also where the package (i.e. '*.cabal' file) is.
+  local project_root = base.match_project_root_pattern(pos.path) or package_root
 
   if lib.files.exists(project_root .. '/cabal.project') then
     return mkCommand(cabal.build_command(package_root, pos))
-  elseif lib.files.exists(project_root .. '/stack.yaml') then
+  elseif lib.files.exists(project_root .. '/stack.yaml') and vim.fn.executable('stack') == 1 then
```

Comment:
We shouldn't check for whether Stack is installed here.

Users who are trying to run this on a Stack project should see an error message if Stack isn't installed (ditto for Cabal).

This could potentially lead to confusing behaviour.

Reply:
Unfortunately, the mere presence of a `stack.yaml` file doesn't mean that it is a stack project. It only means that it *can* be built with stack. Every stack project can be built with cabal, though. 

I agree that this is a hack, but without it, I can't run tests on projects with `neotest-haskell` unless I start using stack. Also other people have strong feelings against stack and don't like using it.

I'll just juggle some ideas around to address this issue, so that it can be solved in another Pull Request:

1. I start using stack when there is a stack.yaml in a project. This might also disappoint other people who also prefer using cabal.

2. A configuration option to prefer cabal for "simple" projects. 

3. Use an environment variable to select cabal for simple projects and we keep the current behavior by default. This has the advantage that opinionated people can simply set it globally and override it when needed with direnv or similar. 

4. Delete the `stack.yaml` and remember not to commit it.

5. I maintain a fork of neotest-haskell with this hack included.

I think the best options are `2` and `3`.

Reply:
I personally don't like using Stack either. But I have it installed. A configuration option would probably be most familiar for Neovim users. Even if it's just something like `vim.g.neotest_haskell_prefer_stack` (which defaults to `false`).

* If it's set to `true`, we use Stack if a `stack,yaml` exists in the project root.
* If it's set to `false`, we use Stack only if a `stack.yaml` exists in the project root and there is no `cabal.project` or `*.cabal`.

We have the same issue for multi-package projects. They can also have a `stack.yaml`, which is why I had checked for the existence of a `cabal.project` file first.

I wouldn't be opposed to using an environment variable if the need for it arises.

Reply:
See also [my updated proposal](https://github.com/mrcjkb/neotest-haskell/pull/44#discussion_r1118168943).

Reply:
Neotest can actually pass a table of config options (see the `vim-test` example) in [readme's configuration section] (https://github.com/nvim-neotest/neotest#configuration).
Perhaps that's a better option than a `vim.g` variable.

But I can understand if that's too much for this PR. I can also implement that myself when I have some time.

Reply:
I look forward to seeing what you come up with 😄 

I've opened #51, which you can reference.